src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -332,7 +332,7 @@ void Default(Exception exception, InvocationContext context)
 
         public static CommandLineBuilder UseHelp(this CommandLineBuilder builder)
         {
-            return builder.UseHelp(new HelpOption());
+            return builder.UseHelp(new HelpOption(builder.Resources ?? Resources.Instance));
```

Comment:
This does raise an interesting question since the value provided here could be different depending on the order that the caller invokes the `.UseResources` and `.UseHelp` extension methods. The same applies to the changes to the Version option below as well.
@jonsequitur feedback?

It seems like we should either explicitly document this behavior, or adjust the internal `.UseHelp` method lazily create the HelpOption.

Reply:
It's important for these methods to be callable in any order and preferably idempotent. I think it's better to avoid passing the `Resources` via `HelpOption`.

Reply:
@Keboo @jonsequitur I made a commit that should resolve the issue. Now the descriptions of the `HelpOption` and version option are set when building the command line parser.